src/core/conversion/viewUtils.test.ts

```diff
@@ -296,6 +356,91 @@ describe('series metadata', () => {
             expect(series1.length).toEqual(1);
             expect(series1[0].precision).toEqual(contentVariable.values[1].contentComponent?.numberOfDecimals);
         });
+
+        it('returns the correct series without selectable variables', () => {
+            const pxGrafResponse: IQueryVisualizationResponse = generatePxGrafResponse([1], [2], [2, 2], [2, 2], []);
+            const selectedValueCodes: TVariableSelections = {};
+            const series: { columnNameGroups: TMultiLanguageString[][], series: IDataSeries[] } =
+                buildSeries(pxGrafResponse, selectedValueCodes);
+            const values: (number | null)[] = series.series.flatMap(s => s.series.map(d => d.value));
+            expect(values).toEqual(pxGrafResponse.data);
+        });
+
+        const pxGrafResponse: IQueryVisualizationResponse = generatePxGrafResponse([1], [2], [2, 2], [2], [2]);
+        const selectableVariableCode: string = pxGrafResponse.selectableVariableCodes[0];
+        const selectableVariableValueCode: string = pxGrafResponse.metaData.find(v => v.code === selectableVariableCode)?.values[1].code as string;
+        const selectedValueCodes: TVariableSelections = { [selectableVariableCode]: [selectableVariableValueCode] };
+        const expectedData: number[] = [17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32];
+
+        it('returns the correct series when there is a selectable variable', () => {
+            const series: { columnNameGroups: TMultiLanguageString[][], series: IDataSeries[] } =
+                buildSeries(pxGrafResponse, selectedValueCodes);
+            const data: (number | null)[] = series.series.flatMap(s => s.series.map(d => d.value));
+            expect(data).toEqual(expectedData);
+        });
+
+        it('returns the same data values in the series after reordering row and column variables', () => {
+            const reordered: IQueryVisualizationResponse = {
+                ...pxGrafResponse,
+                columnVariableCodes: [pxGrafResponse.rowVariableCodes[1], pxGrafResponse.rowVariableCodes[0]],
+                rowVariableCodes: [pxGrafResponse.columnVariableCodes[1], pxGrafResponse.columnVariableCodes[0]]
+            };
+            const series: { columnNameGroups: TMultiLanguageString[][], series: IDataSeries[] } =
+                buildSeries(reordered, selectedValueCodes);
+            const data: (number | null)[] = series.series.flatMap(s => s.series.map(d => d.value));
+            expect(data.sort()).toEqual(expectedData);
```

Comment:
Why are the series combined and sorted? I assumed we want to test that the data is in correct order.

Reply:
That is correct. Removed sorting from the expected results from this and the two following tests.